aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -229,6 +231,7 @@ private SyncTime filterOutOldSyncTimes(SyncTime lastSyncTime) throws DataStoreEx
                         processor.onComplete();
                     }
                 })
+                .retry(retryStrategy::retryHandler)
```

Comment:
If an error occurs syncing page 3 for a given model, will this just retry page 3 with exponential backoff, or is it going to start over with page 1?  Ideally, we want the former.    I think you may need to call `.retry` on the `Single` returned by the `syncPage` method instead, to accomplish this behavior.

Also, we should make sure that we are only retrying retryable errors (non 4xx errors).